### PR TITLE
Open old dot link with openOldDotLink()

### PR DESCRIPTION
**Pull request by mateusbra**
### Details

We are going to use the `openOldDotLink()` function when we have oldDot links, it will open the oldDot link with our account logged in.
### Fixed Issues

$ https://github.com/Expensify/App/issues/7909
$ https://github.com/Expensify/App/issues/8943

### Tests

**Test 1 - old dot links**
1. Launch the app.
2. Open the report with Concierge.
3. Click on the "Request a call" link sent by Concierge.
4. Verify that the link opens in old dot with the user logged in.

**Test 2 - new dot links**
1. Launch the app.
2. Open any report.
3. Send this link - https://new.expensify.com/statements/123
4. Verify that clicking on the link navigates the app to a screen within the app.

**Test 3 - Attachments**
1. Launch the app.
2. Open any report.
3. Send a video, photo and pdf attachment.
4. Verify that there are no regressions related to viewing, sending, downloading etc.

**Test 4 - Other links**
1. Open any report.
2. Send a link eg: http://google.com/
3. The link should open in a new tab.

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

**Test 1 - old dot links**
1. Launch the app.
2. Open the report with Concierge.
3. Click on the "Request a call" link sent by Concierge.
4. Verify that the link opens in old dot with the user logged in.

**Test 2 - new dot links**
1. Launch the app.
2. Open any report.
3. Send this link - https://new.expensify.com/statements/123
4. Verify that clicking on the link navigates the app to a screen within the app.

**Test 3 - Attachments**
1. Launch the app.
2. Open any report.
3. Send a video, photo and pdf attachment.
4. Verify that there are no regressions related to viewing, sending, downloading etc.

**Test 4 - Other links**
1. Open any report.
2. Send a link eg: http://google.com/
3. The link should open in a new tab.

#### Web

https://user-images.githubusercontent.com/56851391/165423526-f2371234-85e5-4862-bc3e-495f039875a0.mov

#### Mobile Web

https://user-images.githubusercontent.com/56851391/165170099-6e40c02d-0166-4dcb-a7a1-89d69c2c9888.mp4

https://user-images.githubusercontent.com/56851391/165441980-ade6421f-0266-46e3-a16e-ecdfb7828b04.mov

#### Desktop

https://user-images.githubusercontent.com/56851391/165440054-e2e6d317-a348-41a2-b1fa-4a87474add98.mov

#### iOS

https://user-images.githubusercontent.com/56851391/165439829-a968cf11-9bb4-44d4-8881-733805e458b1.mov

#### Android

https://user-images.githubusercontent.com/56851391/165169436-8eb6abac-2329-4710-807f-7505347770c2.mp4

### CODE REVIEW

File: src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js [hunk 1]
```diff
@@ -20,16 +22,29 @@ const AnchorRenderer = (props) => {
...
                 style={styles.link}
-                onPress={() => Navigation.navigate(internalExpensifyPath)}
+                onPress={() => Navigation.navigate(internalNewExpensifyPath)}
+            >
+                <TNodeChildrenRenderer tnode={props.tnode} />
+            </Text>
+        );
+    }
+
+    // If we are handling an old dot Expensify link we need to open it with openOldDotLink() so we can navigate to it with the user already logged in.
```
Comment (rushatgabhane) [inline src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js:42]: please update the comment to mention about attachment

**Review by rushatgabhane — COMMENTED** [review]
@mateusbra please change the title of PR to something like 'Open old dot link with openOldDotLink()'

In tests, the video doesn't need to be sent to Concierge. It can be sent to any user.
1. Launch the app
2. Open the report with Concierge <----- this
3. Send a video attachment

File: src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js [hunk 2]
```diff
@@ -20,16 +22,30 @@ const AnchorRenderer = (props) => {
...
+    // As we store attachments under expensify.com URL we don't want it working the same as links.
```
Comment (rushatgabhane) [inline src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js:43]: @mateusbra gonna be nitpicky here

```suggestion
    // As attachments also use expensify.com we don't want it working the same as links.
```

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mateusbra a few issues

Actual: Clicking download on a video on web opens it in a new tab. 
Expected: Video starts downloading.

Actual: On mWeb, clicking download on a video does nothing.
Expected: Video start downloading.

https://user-images.githubusercontent.com/29673073/169721657-8cf129d8-6ba7-45f0-81c9-baff3a53ae37.mp4

**Comment by mateusbra** [comment]
@rushatgabhane I noticed this happenned, I think its a regression from another PR,  it was also reproducible running main on dev, could you confirm that for me?

**Comment by rushatgabhane** [comment]
> I noticed this happenned, I think its a regression from another PR

@mateusbra please report the bug on slack if you haven't already.
<hr />

This bug isn't present on `main`
1. Send a video
2. Right click to open context menu
3. Download

Expected: video downloads
Actual: nothing happens

https://user-images.githubusercontent.com/29673073/169784440-c20c6abd-6a2a-4cb6-ae4e-aeb8813cf8a5.mov

**Comment by mateusbra** [comment]
I wasn't able to reproduce this bug from my side.
This video shows what happens to me running `openOldDotLinksRegression` after merging main(didn't tested this specific bug before merging it), I think its working fine, am I missing something?

https://user-images.githubusercontent.com/56851391/169854283-921408fb-55f5-4447-b298-0a32054a8c8e.mp4

**Review by rushatgabhane — APPROVED** [review]
@iwiznia LGTM! 🎉

<details>
<summary><h4> #### PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Comment by rushatgabhane** [comment]
> I wasn't able to reproduce this bug from my side.

@mateusbra yeah you're right. I can't replicate that bug after you've merged with `main`.
Also, thanks for logging the regression on slack :)

**Comment by rushatgabhane** [comment]
@mateusbra could you please reformat the tests and QA steps to something like this? I think this is more easier to read.

#### Test 1 - old dot links
1. Launch the app.
2. Open the report with Concierge.
3. Click on any link Concierge sends to you.
4. Verify It should open an old dot URL with the user logged in.

#### Test 2- new dot links
1. Open any report
2. Send this link - https://new.expensify.com/statements/123
4. Verify that clicking on the link navigates the app to a screen within the app ().


#### Test 3 - Attachments
1. Open any report.
2. Send a video, photo and pdf attachment.
3. Verify that there are no regressions related to viewing, sending, downloading etc.

#### Test 3 - Other links
1. Open any report.
2. Send a link eg: http://google.com
4. The link should open in a new tab.

**Comment by mateusbra** [comment]
tests updated!

**Review by iwiznia — APPROVED** [review]

**Comment by mateusbra** [comment]
I don't think its a regression, we only changed how `old dot` links works in the PR.

**Comment by iwiznia** [comment]
The timeline also doesn't line up, the [issue](https://github.com/Expensify/App/issues/9158) was opened 2 days ago and says it was reproducible in production, this PR was not in production 2 days ago

**Comment by rushatgabhane** [comment]
> the https://github.com/Expensify/App/issues/9158 was opened 2 days ago and says it was reproducible in production

@iwiznia I think that's a mistake or some misunderstanding

Because J. Nam can repro the issue on staging, but _not_ on production - https://github.com/Expensify/App/issues/9158#issuecomment-1137650295

**Comment by iwiznia** [comment]
Oh... if that's the case, then it is probable this was the cause. @mateusbra can you check it out? Should be easy to test, replicate the bug on main, then try to replicate it with this PR reverted.

**Comment by rushatgabhane** [comment]
@iwiznia our emails (eg: rushatgabhane@gmail.com) don't have access to `wallets` so we always see a white screen. So we can't really test the bug.

**Comment by iwiznia** [comment]
Oh... I don't have access either 😢 
Do you need to actually have a wallet set up or just have access to the feature? Because if it's the latter, I can give you temporary access, if it is the former I probably can't because only people with an SSN from USA can enable the feature.

**Comment by rushatgabhane** [comment]
@iwiznia I have no idea about wallets 😅
@thienlnam can you help us out with https://github.com/Expensify/App/pull/8980#issuecomment-1138424543

**Comment by mateusbra** [comment]
I couldn't reproduce the bug neither on `main` or `this PR reverted` on prod. It navigates me to this blank screen(it doesn't open new tab on browser):
image

**Comment by thienlnam** [comment]
> I don't think its a regression, we only changed how old dot links works in the PR.

The statements page is an OldDot link btw.

Yeah I've tested this on production again, and haven't run into problems so I think this was just a configuration issue on staging that cause it not to work.  

Doesn't seem like there is a regression here

Screen Shot 2022-05-26 at 10 03 46 AM

**Comment by rushatgabhane** [comment]
> Doesn't seem like there is a regression here

Thanks a lot for confirming!

**Pull request merged**